rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -2440,6 +2441,29 @@ int LuaUnsyncedCtrl::CreateDir(lua_State* L)
 	return 1;
 }
 
+/***
+ *
+ * @function Spring.AllocateTable
+ * @number narr hint for count of array elements
+ * @number nrec hint for count of record elements
+ * @treturn table
+ */
+int LuaUnsyncedCtrl::AllocateTable(lua_State* L)
+{
+	int narr = luaL_optinteger(L, 1, 0);
+	int nrec = luaL_optinteger(L, 2, 0);
```

Comment:
Would be good if this only accepted `nil` or integers (so you can't pass 1.23 or "foo" or a function) but I see that "is integer" or "check integer" lualib functions don't actually exist. I guess it can stay as-is then.

Reply:
optinteger does a checkinteger, and you get an error message:
`bad argument #1 to 'AllocateTable' (number expected, got string)`
```c++

LUALIB_API lua_Integer luaL_optinteger (lua_State *L, int narg,
                                                      lua_Integer def) {
  return luaL_opt(L, luaL_checkinteger, narg, def);
}
```

Reply:
hmm maybe not that function though, that one is in lauxlib.cpp

Reply:
Looks like floats are `trunc`d

Reply:
Sounds fine.